cmd/podman/common/completion.go

```diff
@@ -159,76 +216,144 @@ func getNetworks() ([]string, cobra.ShellCompDirective) {
 	return suggestions, cobra.ShellCompDirectiveNoFileComp
 }
 
+// validCurrentCmdLine validates the current cmd line
+// It utilizes the Args function from the cmd struct
+// In most cases the Args function validates the args length but it
+// is also used to verify that --latest is not given with an argument.
+// This function helps to makes sure we only complete valid arguments.
+func validCurrentCmdLine(cmd *cobra.Command, args []string, toComplete string) bool {
+	if cmd.Args == nil {
+		// Without an Args function we cannot check so assume it's correct
+		return true
+	}
+	// We have to append toComplete to the args otherwise the
+	// argument count would not match the expected behavior
+	if err := cmd.Args(cmd, append(args, toComplete)); err != nil {
+		var a, b int
+		var s string
+		// Special case if we use ExactArgs(2) or MinimumNArgs(2),
+		// They will error if we try to complete the first arg.
+		// Lets parse the common error and compare if we have less args than
+		// required. In this case we are fine and should provide completion.
+		_, err = fmt.Sscanf(err.Error(), "%s %d arg(s), received %d", &s, &a, &b)
```

Comment:
`s` is never used; I'm surprised Go doesn't warn about that. And while you're changing this, what would you think of changing `a` and `b` to `need`/`got` or something similar?